### DISCUSSION TITLE
Azure proxy connector - problem CORS (metoda uwierzytelniania)

### DISCUSSION THREAD

**Discussion by odzio33** [AzureCommunityPL/group#201] [Pomoc]
Wprowadziłem zmiany w aplikacji, dodaje teraz w nagłówkach HTTP wpisy związane z Access-Control-Allow ….

HTTP/1.1 200 200
Date: Wed, 27 Jan 2021 11:46:13 GMT
Server: Apache/2.4.37 (centos) OpenSSL/1.1.1c mod_auth_pubtkt/0.13
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Set-Cookie: JSESSIONID=4EEB5CE9898B677E1AB18F8E980C48FC; Path=/atende; Secure; HttpOnly
Content-Type: text/html;charset=ISO-8859-1
Content-Length: 17
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive

Mamy taką sytuacje że gdy na stronie SharePoint-a zamieszcze przycisk który wysyła żądanie do https://XXX/pub/diagnostic.jsp
wszystko działa poprawnie - CORS nie zgłasza błędu  (przed dodaniem wpisów do nagłówków błąd był zgłaszany)

<button onclick="var req = new XMLHttpRequest();
req.open('GET', 'https://XXX/pub/diagnostic.jsp', false);
req.send(null);
if(req.status == 200)
dump(req.responseText);">
Click me
</button>

Jeżeli jednak do tej samej strony wysyłamy żądnie przez Azura na adres https://XXX/pub/diagnostic.jsp
problem CORS nadal występuje – prawdę mówiąc nie wiem dlaczego przeglądarka odrzuca to połączenie, być może powodem jest to, że Azue lub Azure Proxy Conector modyfikuje nagłówek  odpowiedzi (ale my nie mamy na to wpływu)
W tym przypadku nagłówek HTTP wygląda  następująco :

HTTP/1.1 200 200
Content-Length: 17
Content-Type: text/html;charset=ISO-8859-1
x-ms-proxy-app-id: XXXXXXXX-XXXXXXXXX
x-ms-proxy-group-id: 8da0997f-994a-4d9a-adf6-XXXXXXXXXX
x-ms-proxy-subscription-id: db9c4a4f-462a-4ee3-a07f-XXXXXXXXXXXX
x-ms-proxy-transaction-id: bc900c13-9070-435e-ba28-XXXXXXXXXXXX
x-ms-proxy-service-name: proxy-appproxy-WEUR-AMS01P-3
x-ms-proxy-data-center: WEUR
x-ms-proxy-connector-id: d8498e6f-76db-4d72-80ac-XXXXXXXXXXXX
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Nel: {"report_to":"network-errors","max_age":86400,"success_fraction":0.001,"failure_fraction":1.0}
Report-To: {"group":"network-errors","max_age":86400,"endpoints":[{"url":"https://ffde.nelreports.net/api/report?cat=proxy-appproxy-WEUR-AMS01P-3"}]}
Date: Wed, 27 Jan 2021 11:35:24 GMT

Jak widzicie wpisy związane z CORS  „Access-Control-Allow-Origin: *” i „Access-Control-Allow-Credentials: true” są zmieszczone  nagłówku  a mimo to przeglądarka blokuje żądanie.

W ramach testów umieściłem na stronie z listą kontrahentów  (https://XXX/bs/index.jsp?page=KhtList ) skrypt:
<button onclick="var req = new XMLHttpRequest();
req.open('GET', 'https://XXX/pub/diagnostic.jsp', false);
req.send(null);
if(req.status == 200)
dump(req.responseText);">
Click me
</button>

Gdy umieściłem ten sam skrypt bezpośrednio na stronie sharepoint-a odrzuci mnie CORS, a po zablokowaniu CORS w przeglądarce odrzuci na mechanizm autentykacji Azure.

Reasumując, problem nadal nie jest rozwiązany. 
W mojej opinii po stronie aplikacji wewnętrznych wszystko działa prawidłowo. 
Jeśli macie jeszcze jakieś pomysły/propozycje co mógłbym zmienić proszę o informację?! :)

Zaobserwowałem jednak bardzo ciekawe zachowanie, jeśli zmienię wstępny sposób uwierzytelniania mojej aplikacji w konfiguracji Azura z „Acitive Directory” na „Przekazywanie”
wszystko działa właściwe - jednak wtedy nie mogę tak skonfigurować Azure żebyśmy mieli w nagłówkach informację o osobie zalogowanej.
Takie zachowanie potwierdza  że problem jest związany z konfiguracją Azure.

**Comment by PawelHaracz** [comment]
Hej,  

Jak dobrze rozumiem, wasza aplikacja pod adresem `https://XXX/` jest on-premise?  jeśli tak to czy robiłeś tak jak  w linkach poniżej?
https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/application-proxy-understand-cors-issues
https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/application-proxy-wildcard
